source/cell/cell.cpp

```diff
@@ -767,9 +814,115 @@ bool cell::has_format() const
+    if (!workbook().owns_format(new_format))
+    {
+        copy_format_from_other_workbook(new_format);
+        return;
+    }
+
+    // Same workbook: direct assignment (original behavior)
     d_->format_ = new_format.d_;
 }
 
+void cell::copy_format_from_other_workbook(const class format &source_format)
+{
+    // Option B: Automatically clone the format to this workbook's stylesheet
+    // Start with a blank format from our workbook
+    auto cloned_format = workbook().create_format(true);
+
+    // Copy alignment if present
+    if (source_format.d_->alignment_id.is_set())
+    {
+        auto src_alignment = source_format.d_->parent->alignments.at(source_format.d_->alignment_id.get());
+        cloned_format.alignment(src_alignment, source_format.d_->alignment_applied);
+    }
+
+    // Copy border if present
+    if (source_format.d_->border_id.is_set())
+    {
+        auto src_border = source_format.d_->parent->borders.at(source_format.d_->border_id.get());
+        cloned_format.border(src_border, source_format.d_->border_applied);
+    }
+
+    // Copy fill if present
+    if (source_format.d_->fill_id.is_set())
+    {
+        auto src_fill = source_format.d_->parent->fills.at(source_format.d_->fill_id.get());
+        cloned_format.fill(src_fill, source_format.d_->fill_applied);
+    }
+
+    // Copy font if present
+    if (source_format.d_->font_id.is_set())
+    {
+        auto src_font = source_format.d_->parent->fonts.at(source_format.d_->font_id.get());
+        cloned_format.font(src_font, source_format.d_->font_applied);
+    }
+
+    // Copy number format if present
+    if (source_format.d_->number_format_id.is_set())
+    {
+        const auto number_format_id = source_format.d_->number_format_id.get();
+        xlnt::number_format src_number_format;
+
+        if (number_format::is_builtin_format(number_format_id))
+        {
+            src_number_format = number_format::from_builtin_id(number_format_id);
+        }
+        else
+        {
+            const auto it = std::find_if(source_format.d_->parent->number_formats.begin(),
+                source_format.d_->parent->number_formats.end(),
+                [number_format_id](const xlnt::number_format &nf) {
+                    return nf.id() == number_format_id;
+                });
+            if (it != source_format.d_->parent->number_formats.end())
+            {
+                src_number_format = *it;
+            }
+        }
+
+        cloned_format.number_format(src_number_format, source_format.d_->number_format_applied);
+    }
+
+    // Copy protection if present
+    if (source_format.d_->protection_id.is_set())
+    {
+        auto src_protection = source_format.d_->parent->protections.at(source_format.d_->protection_id.get());
+        cloned_format.protection(src_protection, source_format.d_->protection_applied);
+    }
+
+    // Copy other properties
+    if (source_format.d_->pivot_button_)
```

Comment:
This test is not really needed. We can just call `pivot_button`. Advantage is that we do not depend on the default value of `pivot_button_`